src/Node/HTTP/HTTPS.purs

```diff
@@ -0,0 +1,242 @@
+-- | This module defines low-level bindings to the Node HTTPS module.
+
+module Node.HTTP.HTTPS
+  ( createServer
+
+  , SSLOptions
+  , handshakeTimeout
+  , requestCert
+  , rejectUnauthorized
+  , NPNProtocols(..)
+  , npnProtocols
+  , ALPNProtocols(..)
+  , alpnProtocols
+  , sessionTimeout
+  , ticketKeys
+  , PFX(..)
+  , pfx
+  , Key(..)
+  , key
+  , passphrase
+  , Cert(..)
+  , cert
+  , CA(..)
+  , ca
+  , CRL(..)
+  , crl
+  , ciphers
+  , honorCipherOrder
+  , ecdhCurve
+  , DHParam(..)
+  , dhparam
+  , secureProtocol
+  , secureOptions
+  , sessionIdContext
+  ) where
+
+import Prelude
+
+import Control.Monad.Eff (Eff)
+import Data.ArrayBuffer.Types (Uint8Array)
+import Data.Foreign (Foreign, toForeign)
+import Data.Functor.Contravariant (cmap)
+import Data.Options (Options, Option, options, opt)
+import Node.Buffer (Buffer)
+import Node.HTTP (Request, Response, Server, HTTP)
+
+-- | The type of HTTPS server options
+data SSLOptions
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+handshakeTimeout :: Option SSLOptions Int
+handshakeTimeout = opt "handshakeTimeout"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+requestCert :: Option SSLOptions Boolean
+requestCert = opt "requestCert"
+
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+rejectUnauthorized :: Option SSLOptions Boolean
+rejectUnauthorized = opt "rejectUnauthorized"
+
+-- | The npnProtocols option can be a String, a Buffer, a Uint8Array, or an
+-- | array of any of those types.
+data NPNProtocols
```

Comment:
I had originally tried to combine the types that were shared across multiple properties.  But in the end, I decided against it.  There are some option types that I omitted in this implementation, but could be added down the line if anyone needs it.  For instance, `key` can take an `Object[]` with a specific spec.  However, no other options can take that spec.

Someone could come along and have a need to implement this option type, and combining option types would make this very difficult and require breaking changes.  Also, the Node libraries could be extended to take additional option types, and they may not extend the spec consistently across options.  Keeping the option types specific to the options is clear, descriptive, flexible, and future-proof.

And it's not so bad from a user's perspective--users will import only the options they need, and the types corresponding to those options.  For the vast majority of use cases, users would only need to import 1-2 different types.